fixture-monkey-kotlin/src/main/kotlin/com/navercorp/fixturemonkey/kotlin/introspector/DurationIntrospector.kt

```diff
@@ -0,0 +1,53 @@
+/*
+ * Fixture Monkey
+ *
+ * Copyright (c) 2021-present NAVER Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.navercorp.fixturemonkey.kotlin.introspector
+
+import com.navercorp.fixturemonkey.api.arbitrary.CombinableArbitrary
+import com.navercorp.fixturemonkey.api.generator.ArbitraryGeneratorContext
+import com.navercorp.fixturemonkey.api.introspector.ArbitraryIntrospector
+import com.navercorp.fixturemonkey.api.introspector.ArbitraryIntrospectorResult
+import com.navercorp.fixturemonkey.api.matcher.Matcher
+import com.navercorp.fixturemonkey.api.property.Property
+import com.navercorp.fixturemonkey.kotlin.matcher.Matchers
+import kotlin.random.Random
+import kotlin.time.DurationUnit
+import kotlin.time.toDuration
+
+private const val SEED = 234L
+
+class DurationIntrospector : ArbitraryIntrospector, Matcher {
+    private val MATCHER = Matchers.DURATION_TYPE_MATCHER
+
+    override fun match(property: Property?): Boolean {
+        return MATCHER.match(property)
+    }
+
+    override fun introspect(context: ArbitraryGeneratorContext?): ArbitraryIntrospectorResult {
+        val property = context!!.arbitraryProperty
+
+        if (!property.isContainer) {
+            return ArbitraryIntrospectorResult.NOT_INTROSPECTED
+        }
+
+        require(match(context.resolvedProperty)) { "Given type is not Duration type: " + context.resolvedType }
+
+        return ArbitraryIntrospectorResult(
+            CombinableArbitrary.from { Random(SEED).nextLong().toDuration(DurationUnit.NANOSECONDS) }
+        )
+    }
```

Comment:
If you want to convert `long` type to `Duration` type, then it is better to add a `PropertyGenerator` which generates a child property of long type for `Duration`.

Then, get the CombinableArbitrary of long type using `getCombinableArbitrariesByArbitraryProperty` in `ArbitraryGeneratorContext` to convert it to Duration type.

Reply:
I've checked the `PrimaryConstructorArbitraryIntrospector` and the value added to the `Duration` constructor returns a long value, but the init method causes the creation to fail.

The problem was caused by the fact that if it was generated in millis, it should be larger than the value generated in nanos.

I considered a `PropertyGenerator`, but it seemed difficult to control the value of the Property in the generator. I solved the problem using your hints, but it doesn't seem like a good way to do it, so how can I improve it?

Reply:
@this-is-spear 
I think there's a better way instead of handling `PrimaryConstructorArbitraryIntrospector`.
You should think about what happens when users use other `ArbitraryIntropsector` types. For example, `JacksonObjectArbitraryIntrospector`.

You should add an option to add `KotlinDurationIntrospector ` in`KotlinPlugin` to use it.

`Property` is not not related to value.  An object is consisted of related properties.
Since `Duration` itself have a `long` Property, you do not need to add a `PropertyGenerator`.
You will need to manipulate `CombinableArbitrary` in order to satisfy your requirement. Perhaps `filter` would help.

Reply:
Thank you! 
I'll use your hints to think of a better way.